errors.go

```diff
@@ -2,6 +2,29 @@ package fargo
 
 // MIT Licensed (see README.md) - Copyright (c) 2013 Hudl <@Hudl>
 
+import (
+	"fmt"
+)
+
+type unsuccessfulHTTPResponse struct {
+	statusCode    int
```

Comment:
I wrote the struct in order from most important to least important, from required to optional, thinking that the response code is the type's _raison d'être _. I tried to keep the string format consistent with the rest of _fargo_'s error messages. I don't think you're advocating changing the string format.

I could swap the field order. I just wanted you to know that its current arrangement was a deliberate choice.